site/src/job_queue/mod.rs

```diff
@@ -197,20 +197,40 @@ pub fn create_benchmark_jobs(
         ) {
             for backend in backends.iter() {
                 for profile in profiles.iter() {
-                    let job = BenchmarkJob::create_queued(
-                        target,
-                        *backend,
-                        *profile,
-                        benchmark_request.tag().unwrap(),
-                        benchmark_set as u32,
-                    );
-                    jobs.push(job);
+                    tx.conn()
+                        .enqueue_benchmark_job(
+                            request_tag,
+                            &target,
+                            backend,
+                            profile,
+                            benchmark_set as u32,
+                        )
+                        .await?;
+                    // If there is a parent, we create a job for it to. The
+                    // database will ignore it if there is already a job there.
+                    // If the parent job has been deleted from the database
+                    // but was already benchmarked then the collector will ignore
+                    // it as it will see it already has results.
+                    if let Some(parent_sha) = benchmark_request.parent_sha() {
+                        tx.conn()
+                            .enqueue_benchmark_job(
+                                parent_sha,
+                                &target,
+                                backend,
+                                profile,
+                                benchmark_set as u32,
+                            )
+                            .await?;
+                    }
                 }
             }
         }
     }
 
-    Ok(jobs)
+    tx.conn()
+        .update_benchmark_request_status(request_tag, BenchmarkRequestStatus::InProgress)
+        .await?;
+    Ok(())
```

Comment:
The tx commit got lost :)

Reply:
:fearful: woops, I've put it back; 754f63b3488657f542e1f43d33945ff4a2e17565